pkg/multicluster/register/registrant.go

```diff
@@ -594,26 +594,64 @@ func (c *clusterRegistrant) getTokenForSa(
 	if err != nil {
 		return "", err
 	}
-	var foundSecret *corev1.Secret
+
+	var sa *corev1.ServiceAccount
+
 	if err = retry.Do(func() error {
-		sa, err := saClient.GetServiceAccount(ctx, saRef)
+		sa, err = saClient.GetServiceAccount(ctx, saRef)
 		if err != nil {
 			return err
 		}
-		if len(sa.Secrets) == 0 {
+		return nil
+	}, SecretLookupOpts...); err != nil {
+		return "", err
+	}
+
+	// create a secret name format. This is similar to what was previously done
+	// but after the k8s upgrade we no longer have knowledge of what was previously
+	// autogenerated by k8s.
+	// TODO(nfuden): move away from these long lived connections as they are
+	// likely to be deprecated even more in the future.
+	secretName := fmt.Sprintf("%s-secret", sa.Name)
+
+	// see if there is a secret
+	_, err = remoteSecretClient.GetSecret(ctx, client.ObjectKey{Name: secretName, Namespace: saRef.Namespace})
```

Comment:
this should be moved into the retry loop, and why is there a separate call to GetSecret in the retry below?

Reply:
At this point we do not believe there exists a secret already. If there is we should not destroy it by our upsert.
This is why we dont have this in the loop.

The loop get secret is to make sure the upsert and k8s controller update has taken place